### PR TITLE
[GDAL] split julia versions for libcurl

### PR DESCRIPTION
**Pull request by visr**
Using the split PROJ version from #2545.

### CODE REVIEW

**Comment by giordano** [comment]
Ok, don't do anything

**Comment by giordano** [comment]
To be clear, I'm worried about https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/952, but if you do `../bundled` that shouldn't a problem I think?

**Comment by visr** [comment]
Got it, not touching anything :)

The setup here is the same as NetCDF and PROJ, with one extra argument to the `configure` function, `proj_jll_version`, which is set to "700.201" here, but needs to be "700.202" for the julia 1.6 build (though anything PROJ 7.2 is ok, see the comment above `proj_jll_version`).

File: G/GDAL/common.jl [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    upstream_version = v"3.2.1"
+    version = VersionNumber(upstream_version.major * 100 + version_offset.major,
+                            upstream_version.minor * 100 + version_offset.minor,
+                            upstream_version.patch * 100 + version_offset.patch)
+
+    # Collection of sources required to build GDAL
+    sources = [
+        ArchiveSource("https://github.com/OSGeo/gdal/releases/download/v$upstream_version/gdal-$upstream_version.tar.gz",
+            "43d40ba940e3927e38f9e98062ff62f9fa993ceade82f26f16fab7e73edb572e"),
+        DirectorySource("./bundled"),
```
Comment (giordano) [inline G/GDAL/common.jl:19]: Ok, I can confirm using `../bunlded` is innocuous:
```suggestion
        DirectorySource("../bundled"),
```

**Pull request merged**